lib/avromatic/model/configuration.rb

```diff
@@ -17,24 +17,30 @@ class Configuration
       # @param options [Hash]
       # @option options [Avro::Schema] :schema
       # @option options [String, Symbol] :schema_name
+      # @option options [String, Symbol] :schema_subject
       # @option options [Avro::Schema] :value_schema
       # @option options [String, Symbol] :value_schema_name
+      # @option options [String, Symbol] :value_schema_subject
       # @option options [Avro::Schema] :key_schema
       # @option options [String, Symbol] :key_schema_name
+      # @option options [String, Symbol] :key_schema_subject
       # @option options [Avromatic::ModelRegistry] :nested_models
       # @option options [Boolean] :mutable, default false
       # @option options [Boolean] :allow_optional_key_fields, default false
       def initialize(**options)
         @avro_schema = find_avro_schema(**options)
+        @avro_schema_subject = options[:schema_subject] || options[:value_schema_subject]
```

Comment:
Would it make sense to default the subject to the schema's full name here to avoid defaulting in a few other places that could get out of sync?

Reply:
Were you thinking of something like `@avro_schema_subject = options[:schema_subject] || options[:value_schema_subject] || avro_schema.fullname`?

This seems to be problematic if the schema is a `Avro::Schema::UnionSchema` since it does not have a `fullname`.
I got this failure `rspec ./spec/avromatic/model/builder_spec.rb:257 # Avromatic::Model::Builder model generation top-level union raises an error`.

Also, I think this only helps with the defaulting in the method `Registration#register_schema`. Since the messaging no longer has the model context. I don't know if this defaulting is worthwhile.